processing/src/main/java/io/druid/query/search/search/ContainsSearchQuerySpec.java

```diff
@@ -0,0 +1,102 @@
+package io.druid.query.search.search;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+
+/**
+ */
+public class ContainsSearchQuerySpec implements SearchQuerySpec
+{
+
+  private static final byte CACHE_TYPE_ID = 0x1;
+
+  private final String value;
+  private final boolean caseSensitive;
+
+  private final String target;
+
+  @JsonCreator
+  public ContainsSearchQuerySpec(
+      @JsonProperty("value") String value,
+      @JsonProperty("caseSensitive") boolean caseSensitive
+  )
+  {
+    this.value = value;
+    this.caseSensitive = caseSensitive;
+    this.target = value == null || caseSensitive ? value : value.toLowerCase();
+  }
+
+  @JsonProperty
+  public String getValue()
+  {
+    return value;
+  }
+
+  @JsonProperty
+  public boolean isCaseSensitive()
+  {
+    return caseSensitive;
+  }
+
+  @Override
+  public boolean accept(String dimVal)
+  {
+    if (dimVal == null) {
+      return false;
+    }
+    final String input = caseSensitive ? dimVal : dimVal.toLowerCase();
+    return input.contains(target);
```

Comment:
Since we are improving this code anyway, I think we should switch to using [StringUtils.containsIngoreCase](https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#containsIgnoreCase(java.lang.String, java.lang.String)) instead of toLowerCase+contains, since lowercasing a string is not always the same for non-ascii characters.